client/signup/steps/design-type/index.jsx

```diff
@@ -28,6 +28,14 @@ class DesignTypeStep extends Component {
 		translate: identity
 	};
 
+	choiceHandlers = {};
+
+	componentWillMount() {
+		this.choiceHandlers = transform( this.getChoices(), ( handlers, choice ) => {
+			handlers[ choice.type ] = ( event ) => this.handleChoiceClick( event, choice.type );
+		}, {} );
+	}
```

Comment:
Actually, _if_ we were to use `memoize` to make this function-call-based rather than object-property-access-based, the following much smaller alternative would work:

```js
getChoiceHandler = memoize( ( choiceType ) =>
  ( event ) => this.handleChoiceClick( event, choiceType ) )
```